src/config.rs

```diff
@@ -88,8 +103,18 @@ pub struct Config {
 
     /// The cache directory in which to store mail files while they are being downloaded. The
     /// default is operating-system specific.
+    #[serde(default = "default_cache_dir")]
+    pub cache_dir: PathBuf,
+
+    /// The location of the mail dir, where downloaded email is finally stored. If not given,
+    /// mujmap will try to figure out what you want. You probably don't want to set this.
+    #[serde(default = "Default::default")]
+    pub mail_dir: Option<PathBuf>,
+
+    /// The directory to store state files in. If not given, mujmap will try to choose something
+    /// sensible. You probably don't want to set this.
     #[serde(default = "Default::default")]
-    pub cache_dir: Option<PathBuf>,
+    pub state_dir: Option<PathBuf>,
```

Comment:
It seems like `state_dir` shouldn't ever leave `Config::from_path` with a value of `None`, and its optional-ness is only for the sake of serde. I think it's fine for this PR, but please add a `TODO` here about how this should be refactored into a non-`Option` somehow, someday.

Reply:
Done. Yeah, this is one of those things where the config file as a pleasing user interface doesn't match up with the harsh reality of the type system. I have Strong Opinions about this sort of thing from my life as a sysadmin, but I will spare you that rubbish.